Jaden Taking Over For Peterson: Improved The Appearance Of The Loading Message On Permissions Page (DONE Jaden)

peterson337 opened the pull request:
Jaden Taking Over For Peterson: Improved The Appearance Of The Loading Message On Permissions Page (DONE Jaden) #3854

# Description
This open PR was created to improve the loading message on the permissions page.

## Related PRS (if any):
None.

## Main changes explained:
The PermissionsManagement.jsx component has been modified to improve the loading message.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. Log in as an admin or owner user
5. go to Other Links → Permissions Management → Manage User Permissions
6. This page has a table, but before the table appears, a loading message will be shown. Check if the message text is white in dark theme or black in light theme. Furthermore, the message needs to be centered on the screen.

## Screenshots or videos of changes:
<img width="1689" height="937" alt="Screenshot 2026-07-03 at 11 43 28 PM" src="https://github.com/user-attachments/assets/bbf3e6e1-b99f-45f8-89d5-91549c6590ed" />

## Note:
None.

kanishkagarwal6101 reviewed (approved):
Great I can see the new loading spinner is looking good is centered and works well. The code is also clean and the class names are not generic.

Jaden300 committed:
Improve loading message appearance on Permissions Management page

Wrap loading text in a centered flex container alongside a spinner,
and apply dark mode text color via the existing darkMode prop.

Jaden300 committed:
Fix loading message test to use data-testid after Spinner was added

kzou55 reviewed (approved):
Hi Jaden, 

I ran and tested the PR locally. 

## Verified

- The loading message alongside the spinner is present and in black text
    <img width="1909" height="1039" alt="Image" src="https://github.com/user-attachments/assets/7aea7cb5-5a97-4855-b838-9af30849f4ac" />

- Mobile View
    <img width="707" height="969" alt="Image" src="https://github.com/user-attachments/assets/3f716ae9-c164-42d4-9bce-c2dbeeeaf7f4" />

- Looks good in Dark Mode(white text)

    <img width="936" height="678" alt="Image" src="https://github.com/user-attachments/assets/fe5e732f-23e9-488f-bccb-da67d79654ac" />

Note: The testing instructions from the PR is a bit outdated as it tells to go to "Manage User Permissions" when the actual log table is below the User Roles component. And you need to be an owner account to see the log table.
